app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -46,14 +47,31 @@ def raw_delete_in_provider
   end
 
   def git_repository
-    super || begin
-      transaction do
-        update!(:git_repository => GitRepository.create!(:url => scm_url))
+    (super || (ensure_git_repository && super)).tap { |r| sync_git_repository(r) }
```

Comment:
Doesn't `ensure_git_repository` return a boolean if it does an update? I don't think we want the return value of `git_repository` to be boolean, right?

Reply:
Right, that's why it does `&& super`, which ensure that's it will return whatever super returns.

Reply:
Right, but if `ensure_git_repository` returns `false` we will never run `super` because `&&` will short-circuit.